scvi/model/base/_utils.py

```diff
@@ -14,13 +14,84 @@
 
 from scvi.data._constants import _SETUP_METHOD_NAME
 from scvi.data._download import _download
+from scvi.module.base._decorators import _move_data_to_device
 from scvi.utils import track
 
 from ._differential import DifferentialComputation
 
 logger = logging.getLogger(__name__)
 
 
+def subset_distribution(my_distribution, index, dim=0):
+    """Utility function to subset the parameter of a Pytorch distribution."""
+    return my_distribution.__class__(
+        **{
+            name: torch.index_select(
+                getattr(my_distribution, name), dim=dim, index=index
+            )
+            for name in my_distribution.arg_constraints.keys()
+        }
+    )
+
+
+def move_distribution(my_distribution, device):
+    """Utility function to move the parameter of a Pytorch distribution to a specific device."""
+    new_params = {
+        name: _move_data_to_device(getattr(my_distribution, name), device)
+        for name in my_distribution.arg_constraints.keys()
+    }
+    return my_distribution.__class__(**new_params)
+
+
+class DistributionsConcatenator:
```

Comment:
this should go in scvi.distributions

Reply:
shouldn't there be a way to find the batch dimension of the distribution? and make this more general?

Reply:
or are we constraining this to distributions with 1 batch dim

Reply:
we also need to test this